addon/bootstrap.js

```diff
@@ -125,6 +180,10 @@ function start(webExtension) {
 
 function stop(webExtension, reason) {
   webExtension.shutdown(reason);
+
+  if (reason != APP_SHUTDOWN) {
```

Comment:
It's just not necessary - all `uninit` does is remove a bunch of DOM nodes, but if we're shutting down all the windows go away so there's no point. Part of the quantum/photon work is making sure startup and shutdown are as quick as possible, so I avoid doing the busy-work here.

It might be that some of the webextension shutdown stuff also doesn't need to happen if we're shutting down, but I haven't examined that in detail and it's kind of orthogonal to the rest of this PR.